### DISCUSSION TITLE
Nesting Templates - Quickadd vd Templater issue

### DISCUSSION THREAD

**Discussion by aubreyz** [chhoumann/quickadd#201] [Q&A]
There must be a way around this. 

- I have a template file with my standard YAML header file (which happens to use some Templater code, which is fine because Quickadd uses this)
- The same Standard YAML header is used as part of all my header templates so I include that using standard format

So my Quickadd Template for an article type might be

```
{{TEMPLATE:Utilities/Template/myyaml.md}}

Some other header data to do with my article type
```

So far so good --- I can have several different templates but only ever need to edit the YAML in one place. 

**The problem**

I can create new files with the above, but sometimes I need to enter the header into an existing file. I can do this with Templater. The problem is that although Quickadd uses Templater Syntax, this does NOT extend to properly using the standard Templater syntax for file nesting

So if I have the below template file (without the exact path), it is usable by Templater but not by Quickadd

```
<% tp.file.include("[[myyaml]]") %>

Some other header data to do with my book type
```

This means having to maintain two entirely separate sets of templates
a) For Quickadd to create new files in a nice way
b) For Templater to add the same headers to existing files

Surely there must be some way around this - either to have common code that can be used by both Plugins (as it would be if I wasn't nesting the templates) or to have Quickadd adding a header to an existing file. 

Any hints please?

**Comment by chhoumann** [comment]
Coming back to this very late - sorry about that.

There's a QuickAdd-native way to keep one template set for both jobs now - no `tp.file.include` needed for the existing-file case.

`{{TEMPLATE:...}}` also works as a **capture format**, and captures can write into the active file. So:

1. Create a Capture choice and enable *Capture to active file*.
2. Set the capture format to the same include your Template choices already use:

```
{{TEMPLATE:Utilities/Template/myyaml.md}}

Some other header data to do with my article type
```

3. Pick the write position that suits you - *Top of file (after frontmatter)* is usually right for a header, or *At cursor* if you want it where Templater used to drop it.

QuickAdd inserts the template file's contents and then runs the usual formatting passes on the result, so tokens nested inside `myyaml.md` resolve too. One `myyaml.md` now serves both the Template choices (new files) and this Capture choice (existing files).

If `myyaml.md` still has Templater snippets in it, I'd convert those to QuickAdd tokens while you're at it (`{{DATE:YYYY-MM-DD}}`, `{{VALUE:...}}`, etc.) so both paths are pure QuickAdd and you never depend on when Templater fires.

Docs: https://quickadd.obsidian.guide/docs/FormatSyntax#template

Closing as resolved, but shout if your setup hits an edge this doesn't cover.
